src/libstd/sys/windows/process.rs

```diff
@@ -141,12 +138,34 @@ impl Command {
                                    .with_extension(env::consts::EXE_EXTENSION);
                     if fs::metadata(&path).is_ok() {
                         return Some(path.into_os_string())
+                    } else {
+                        // Windows relies on path extensions to resolve commands.
+                        // Path extensions are found in the PATHEXT environment variable.
+                        if let Some(exts) = env_pathext {
+                            for ext in split_paths(&exts) {
+                                let ext_str = ext.to_string_lossy();
```

Comment:
I think we'll want to call `to_str` here to not misinterpret OS strings (e.g. lossy may change the meaning), but you can just skip anything that's `None`. You can probably do:


```
for ext in split_paths(&ext).filter_map(|e| e.to_str()) {
    // ...
}
```